aws/eks/eks-how-to/deploy-ubuntu-pro-cluster.rst

```diff
@@ -0,0 +1,282 @@
+~~~~~~~~~~~~~
+
+You need:
+
+- ``eksctl``: see how to install it `here <https://docs.aws.amazon.com/eks/latest/userguide/eksctl.html>`_
+- ``packer``: only needed if you want to enable FIPS for the cluster nodes. Install it with ``sudo snap install packer``
+- your AWS access key ID and secret access key
+- an Ubuntu Pro token
+
+
+Cluster deployment preparation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Depending on whether you need to enable FIPS or not, you only need to follow one of the following two sections.
+
+Please note that, while there are Ubuntu Pro AMIs available in AWS, at the time
+of writing this guide, there is no such offering for the EKS service, so you'll
+need to provision the EKS cluster with customized Ubuntu nodes.
+
+
+With FIPS
+^^^^^^^^^
+
+When enabling FIPS, a reboot of the underlying node is required, which if done post-cluster creation, can lead to a defective cluster.
+For this reason, the most reliable way to deploy an Ubuntu Pro EKS cluster is to actually build a
+custom Ubuntu Pro AMI (with `Packer <https://www.packer.io/>`_) and use it at cluster creation time.
+
+ 
+What are the caveats?
+*********************
+
+You need to keep rebuilding your custom Ubuntu Pro image in order to have the
+latest updates and security fixes from upstream.
+
+Also, storing AMI has a cost on AWS, and you will have to replicate it to multiple regions if you need to.
+
+Get and modify the Packer file
+******************************
+
+Create an ``eks-ubuntu-fips.json`` file with the following (replacing the
+placeholder credentials with your own in the "variables" section):
+
+..  code-block:: bash
+
+    {
+        "variables": {
+            "aws_access_key": "YOUR_IAM_ACCESS_KEY",
+            "aws_secret_key": "YOUR_IAM_SECRET_KEY",
+            "ua_token": "YOUR_UA_TOKEN",
+            "eks_ver": "1.23"
+        },
+        "builders": [
+        {
+            "type": "amazon-ebs",
+            "access_key": "{{user `aws_access_key`}}",
+            "secret_key": "{{user `aws_secret_key`}}",
+            "region": "us-east-1",
+            "instance_type": "t2.micro",
+            "ami_name": "eks{{user `eks_ver`}}-fips-ubuntu20.04-{{timestamp}}",
+            "source_ami_filter": {
+                "filters": {
+                    "virtualization-type": "hvm",
+                    "name": "ubuntu-eks/k8s_{{user `eks_ver`}}/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-*",
+                    "root-device-type": "ebs"
+                },
+            "owners": ["099720109477"],
+            "most_recent": true
+        },
+        "ssh_username": "ubuntu"
+        }
+        ],
+        "provisioners": [
+        {
+            "type": "shell",
+            "inline": [
+            "cloud-init status --wait",
+            "sudo apt-get update && sudo apt-get upgrade -y --with-new-pkgs"
+            ]
+        },
+        {
+            "type": "shell",
+            "inline": [
+            "sudo ua attach {{user `ua_token`}}",
```

Comment:
is that token somewhere removed? because if not, we reuse the same token on all machines which are using this golden image.

Reply:
On the packer file, we remove the machine-id, if I create a cluster with two nodes, it's will be counted as 2 Ubuntu pro subscription.